src/features/archaeologist/hooks/useLoadArchaeologists.ts

```diff
@@ -1,20 +1,38 @@
+import { LibP2pContext } from 'lib/network/P2PNodeProvider';
+import { useContext } from 'react';
+import { startLoad, stopLoad } from 'store/app/actions';
 import { useAsyncEffect } from '../../../hooks/useAsyncEffect';
 import { useDispatch } from '../../../store';
 import { storeArchaeologists } from '../../../store/archaeologist/actions';
 import { Archaeologist } from '../../../types';
+import { initialisePeerDiscovery } from '../discovery';
 
 export function useLoadArchaeologists() {
   const dispatch = useDispatch();
 
+  const browserNode = useContext(LibP2pContext);
+
   useAsyncEffect(async () => {
     try {
+      dispatch(startLoad());
       const archaeologists: Archaeologist[] = [];
-      // TODO: Add function to load archaeologists
+
+      if (browserNode === undefined) {
+        console.error('browser node is undefined');
+        return;
+      }
+
+      await initialisePeerDiscovery(
+        await browserNode,
+        (discoveredArchs: Archaeologist[]) => dispatch(storeArchaeologists(discoveredArchs))
+      );
 
       dispatch(storeArchaeologists(archaeologists));
     } catch (error) {
       // TODO: Implement better error handling
       console.error(error);
+    } finally {
+      dispatch(stopLoad());
```

Comment:
Unfortunately the loading circle won't work with the way the archs are being loaded. We will have to set up a reducer action that detects when the archs are finished loading... In a future task of course.

Reply:
Along these lines -- there may not be a hard "stop" when the archaeologists have finished loading. The browser will (most likely) have its peer discovery mechanism open for the duration of the browser session, and archaeologists can continue to be discovered (or disconnected) at any point.

We could manually enforce a time limit or a # of archaeologists found (i.e. 5) before showing the user results.

Another thing to consider -- we will most likely be doing on-chain archaeologist profiles. So we could opt to start with the list of all archaeologists that are registered on-chain, and then show them as active/inactive as the browser node discovers these peers in the background.

Reply:
Yeah I like the idea of enforcing a minimum archaeologist count before showing results.